### PR TITLE
For #15403 - Adds a secret debug info screen in settings

### PR DESCRIPTION
**Pull request by boek**
image
This will show the device/user id if they exist

### CODE REVIEW

**Review by Amejia481 — APPROVED** [review]
🚢

File: app/src/main/java/org/mozilla/fenix/settings/SecretDebugSettingsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import android.os.Bundle
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import com.leanplum.Leanplum
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.showToolbar
+
+/**
+ * Lets the user customize Private browsing options.
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/SecretDebugSettingsFragment.kt:15]: Could you add a bit more here? I don't think this description matches my understanding of what this settings menu is doing

**Review by eliserichards — CHANGES_REQUESTED** [review]
Just one small nit :)

**Comment by boek** [comment]
@eliserichards nits have been fixed, just need the green check :)

File: app/src/main/java/org/mozilla/fenix/settings/SecretDebugSettingsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        showToolbar(getString(R.string.preferences_debug_info))
+    }
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.secret_info_settings_preferences, rootKey)
+
+        requirePreference<Preference>(R.string.pref_key_leanplum_user_id).apply {
+            summary = Leanplum.getUserId().let {
+                if (it.isNullOrEmpty()) {
+                    "No User Id"
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/SecretDebugSettingsFragment.kt:27]: Should we add these strings to static_strings as well?

File: app/src/main/java/org/mozilla/fenix/settings/SecretDebugSettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                } else {
+                    it
+                }
+            }
+        }
+
+        requirePreference<Preference>(R.string.pref_key_leanplum_device_id).apply {
+            summary = Leanplum.getDeviceId().let {
+                if (it.isNullOrEmpty()) {
+                    "No Device Id"
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/SecretDebugSettingsFragment.kt:37]: Same as above.

**Review by gabrielluong — APPROVED** [review]
Just wanted to bump this! I am sure @st3fan will be happy to see this land!

**Comment by st3fan** [comment]
Super happy :-)

**Review by eliserichards — APPROVED** [review]
Just fixed conflicts with the latest `master`. Once CI is green this can land and we can make @st3fan happy 😄

**Pull request merged**